File: testdata/jennies/rawtypes/disjunctions_of_refs_without_discriminator/JavaRawTypes/disjunctions_of_refs_without_discriminator/TypeA.java
package disjunctions_of_refs_without_discriminator;

import java.util.Objects;

public class TypeA {
    public String fieldA;
    public TypeA() {
        this.fieldA = "";
    }
    public TypeA(String fieldA) {
        this.fieldA = fieldA;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof TypeA)) return false;
        TypeA o = (TypeA) other;
        if (!Objects.equals(this.fieldA, o.fieldA)) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.fieldA);
    }
}

File: testdata/jennies/rawtypes/disjunctions_of_refs_without_discriminator/PHPRawTypes/src/DisjunctionsOfRefsWithoutDiscriminator/TypeA.php
<?php

namespace Grafana\Foundation\DisjunctionsOfRefsWithoutDiscriminator;

class TypeA implements \JsonSerializable
{
    public string $fieldA;

    /**
     * @param string|null $fieldA
     */
    public function __construct(?string $fieldA = null)
    {
        $this->fieldA = $fieldA ?: "";
    }

    public function equals(mixed $other): bool
    {
        if (!($other instanceof self)) {
            return false;
        }
    
        if ($this->fieldA !== $other->fieldA) {
            return false;
        }
    
        return true;
    }

    /**
     * @param array<string, mixed> $inputData
     */
    public static function fromArray(array $inputData): self
    {
        /** @var array{fieldA?: string} $inputData */
        $data = $inputData;
        return new self(
            fieldA: $data["fieldA"] ?? null,
        );
    }

    /**
     * @return mixed
     */
    public function jsonSerialize(): mixed
    {
        $data = new \stdClass;
        $data->fieldA = $this->fieldA;
        return $data;
    }
}
